### ISSUE TITLE
Question about the normalized coordinates and img_aug

### ISSUE THREAD

**Issue by Bear-kai** [rpng/suo_slam#1]
- Q1: In the data loader func where makes the ground truth kp_uv [link1](https://github.com/rpng/suo_slam/blob/5de01433d177fde5cac4423f05fd554e3c00794e/lib/datasets/bop.py#L654), coordinate (-1,-1) corresponds to the left-bottom bbox corner.
However, in the mesh_grid func which is used for the predicted kp_uv [link2](https://github.com/rpng/suo_slam/blob/5de01433d177fde5cac4423f05fd554e3c00794e/lib/models/pkpnet.py#L25), coordinate (-1,-1) corresponds to the right-upper bbox corner.
I think they should use the same "starting point", either left-bottom, or right-upper. Do I understand correctly?

- Q2: I found the bboxes may have negative values, and (x1,y1) can be larger than (x2,y2) due to the possible 180 degree rotation in "NpScaleAndRotate". This will cause undesirable behaviour in the "torchvision.ops.roi_align" process ?! At least I have found the cropped image might be like the meaningless one below. Then I doubt is it suitable to do such augmentation?
![ttmmpp_im2_1](https://user-images.githubusercontent.com/10415005/223375975-ad6ca48f-4504-4b81-b037-25b88df829f0.png)

**Comment by nmerrill67** [comment]
I agree that I never liked the normalized coordinates moving the origin out of the top left. It was just a trick I borrowed from the old KeyPointNet repo, and it could probably be fixed easily (I only use that property when transforming from normalized to unnormalized or v.v.). I only manually tuned the augmentations a bit and did not guarantee that the object will always be visible I guess.

**Comment by Bear-kai** [comment]
@nmerrill67 Hi, thanks for your reply. I have another question and expect your help.
I noticed that the evaluation results are based on a final global BA optimization [here](https://github.com/rpng/suo_slam/blob/5de01433d177fde5cac4423f05fd554e3c00794e/evaluate.py#L256). Then, the frames can utilize info from the future. It confuses me cause I think in slam mode, we can only use info of the current frame and from history, not from the future!
